### DISCUSSION TITLE
Copilot doesn’t work if profanity exists

### DISCUSSION THREAD

**Discussion by Nevrai** [community/community#55630] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I’m unsure whether this is truly a bug, but it is frustrating. Whenever there’s profanity in a file, Copilot refuses to work.

If I’m ‘far enough’ away from the piece of code or comments that contains expletives, Copilot will work in that file, and it’ll also work in other files, but it won’t work _near_ the expletives, which is presumably due to the context length limit.

It’s quite bizarre, especially since Copilot _is_ capable of generating profanities itself. I don’t understand why it decides not to work if it’s close to profanity.

**Comment by SamWarden** [comment]
It also doesn't work at all if the project directory name contains profanity #55378

**Reply by ggoonnzzaallo** [reply]
AI safety advocates have gone too far... I built a Copilot VS Code extension that handles all kinds of profanity, you can try it [here](https://marketplace.visualstudio.com/items?itemName=doublebot.doublebot) if you're still looking for a solution. Will never censor anything, that's lame.

**Comment by AndreasOM** [comment]
We have been intentionally spreading profanity through our code bases to keep co-pilot from stealing. It has been very successful so far, and I hope it stays this way.

It's not a bug. It's a feature.

**Reply by axefrog** [reply]
That's a silly way to keep an unexpected benefit as a feature. If they want that to be a feature, they should have well-defined tokens you can use for that purpose. The use of profanity in a codebase is a personal choice and should not have any impact on Copilot's functionality.

**Reply by Lamby777** [reply]
> That's a silly way to keep an unexpected benefit as a feature. If they want that to be a feature, they should have well-defined tokens you can use for that purpose. The use of profanity in a codebase is a personal choice and should not have any impact on Copilot's functionality.

🤓

**Comment by synap5e** [comment]
Just spent time debugging copilot not working because I had shorthanded an "assembly" variable to "ass" in a comment. I love the future.

**Reply by bmod** [reply]
Be sure to "unalive" your processes when they get "unparented"

**Comment by fireattack** [comment]
I was working on a small project which is for processing ASS (Advanced SubStation Alpha) subtitle (https://en.wikipedia.org/wiki/Subtitles#Subtitle_formats) and copilot refused to work due to variable name containing "ass".

This is ridiculous, ass as a word or shorthand is too common to be put into "profanity words list".

**Reply by andraz** [reply]
Refusing to work for paying customers because they're using an acronym to name the folder of After Sale Service library in their application is beyond insane.

What is next, everyone who "dare" to develop anything related to Advanced Sound System will need to use full name to not trigger the idiocy?

It gets worse, try generating some trivial user sample data, it will just stop working when it's time to generate an example@example.com email...

**Reply by Lamby777** [reply]
Yep. It also freaks the fuck out when you try to generate anything containing an IP address, even if it's a local one like 192.168.x.x or 10.x.x.x

Nuts.

**Reply by cartoonandtravel** [reply]
The same problem (」°ロ°)」

**Comment by ggoonnzzaallo** [comment]
Found this pretty funny lol, but strongly believe that you should be able to have anything you want in your code... 

Microsoft doesn't appear to have changed its mind on this 1 year after so I built my own Copilot extension that handles profanities (there are no words banned). You can try it [here](https://marketplace.visualstudio.com/items?itemName=doublebot.doublebot) if you're still looking for a solution.

**Reply by tendiesfortwo** [reply]
Looks good, checking it out. Please never censor.

**Reply by Lamby777** [reply]
Is there a (neo)vim plugin version?

**Reply by nikhilcharan-dev** [reply]
So Cool.

**Comment by nikhilcharan-dev** [comment]
The reported behavior of Copilot, where it refuses to function properly in the vicinity of profanity within code files, is indeed concerning and frustrating. It's understandable for developers to encounter challenges with profanity filtering in code-related tools, but the described behavior seems inconsistent and puzzling.

**Comment by gradientjanitor** [comment]
Just discovered this. The irony is that when I start putting profanity in my code, it is because I am frustrated, then copilot stops working, which frustrates me further.

**Comment by fungiboletus** [comment]
While the completion quality doesn't seem to match Github Copilot yet, [CodeGemma 2B](https://ai.google.dev/gemma/docs/codegemma) has no issues with profanities.

I use it locally with [Continue.dev](https://www.continue.dev) and [Ollama](https://ollama.com).

```json
  "tabAutocompleteModel": {
    "title": "Code Gemma",
    "provider": "ollama",
    "model": "codegemma:code"
  },
 ```
 
Hopefully the existence of good and free models will make Github and Microsoft change their stance towards their quite excessive guardrails.

**Comment by SethRobinson** [comment]
Wasted 20 minutes trying to figure out why copilot wouldn't work... it was the print("Hello ass!"); statement, something I tend to type instead of hello world.  Great.  In addition to the horrible decision to be this sensitive, why on earth would it be a totally silent error so it just seems broken?!

**Comment by jcredland** [comment]
This is absurd. I’ve written an entire function just to check if my data is screwed, and Co-pilot won't touch it. Is this some kind of American prudishness where they think swearing will cause the apocalypse or universal healthcare? The English swear constantly, and the Australians practically use it as punctuation.

**Comment by not-nullptr** [comment]
＞llm
＞large LANGUAGE model
＞​not allowed to output language
????

**Comment by ulfkri** [comment]
I had the same issue and spent hours debugging before I came here.

To identify the offending word, I typed something at the start of the file where completions worked, then moved line by line until completions stopped working. In my case, the issue was caused by the Norwegian word "slutt" (which means "end") in a comment.

What a useless feature. Least they could do was log something at least. Only clue in the log is this: `model deployment ID: []`

**Comment by cartoonandtravel** [comment]
This problem has not been solved for the second year. I'm thinking of canceling my subscription and looking for another solution

**Comment by bordplate** [comment]
This is still an annoying bug. It seems to trigger on extremely mild profanity, and words that aren't considered profanity in my locale.

**Comment by Lamby777** [comment]
Is this just gonna be ignored forever? I'm still having this issue, this time while trying to use Copilot in a Vue component about an album which contains the word "damn" in one of the b-side song titles. (not even in a profane way, just a reference to a somewhat silly mnemonic)

Are we really censoring "damn" now? What is this, the 80s/90s? What could an LLM say using "damn" that couldn't also be said using either "screw" or "stupid?" Who is this supposed to protect?

**Comment by Sv443** [comment]
This royally pisses me off. I PAY for this software and I PAY for it to do as I ask. I don't pay for censorship. I really *fucking* love this future where I have to base64-encode abbreviations because our corpo-overlords are scared of reality.

**Comment by Michal-Zablocki** [comment]
+1, absolutely mind-boggling

**Comment by artlung** [comment]
+1

**Comment by Lamby777** [comment]
Oh the notification reminded me, honestly I've just stopped using "autocomplete" LLMs entirely for a bit. Using a local uncensored one is probably fine, but my hardware is too old for that at any reasonable speed. For the "boilerplate" that I would typically use copilot for, like formatting data into strings or imports and other crap, I still use Gemini CLI. It feels separate enough from my actual editing process that programming feels more fun again. Also not nearly as "integrated" into the workflow-- so rare that I don't feel like I'd be "missing out" if I decide to type some string that Big Tech doesn't like and lost access to it.

Not to mention IT DOESN'T HAVE THIS STUPID ISSUE BUILT INTO IT!!!!!!!!
Here it is, objectively being more capable tool than Copilot. Give it a try-- it's free. 🤷‍♀️

```
refcherry@InterSHELL:~/Code/scrobble-fixer % gemini

 ███            █████████  ██████████ ██████   ██████ █████ ██████   █████ █████
░░░███         ███░░░░░███░░███░░░░░█░░██████ ██████ ░░███ ░░██████ ░░███ ░░███
  ░░░███      ███     ░░░  ░███  █ ░  ░███░█████░███  ░███  ░███░███ ░███  ░███
    ░░░███   ░███          ░██████    ░███░░███ ░███  ░███  ░███░░███░███  ░███
     ███░    ░███    █████ ░███░░█    ░███ ░░░  ░███  ░███  ░███ ░░██████  ░███
   ███░      ░░███  ░░███  ░███ ░   █ ░███      ░███  ░███  ░███  ░░█████  ░███
 ███░         ░░█████████  ██████████ █████     █████ █████ █████  ░░█████ █████
░░░            ░░░░░░░░░  ░░░░░░░░░░ ░░░░░     ░░░░░ ░░░░░ ░░░░░    ░░░░░ ░░░░░


Tips for getting started:
1. Ask questions, edit files, or run commands.
2. Be specific for the best results.
3. Create GEMINI.md files to customize your interactions with Gemini.
4. /help for more information.


╭────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│  > Please replace the word "FUCK" in @CENSORSHIP-SUCKS.sh with "SHIT", and then run it in zsh. Thank you.  │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────╯

 ╭─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
 │ ✔  ReadManyFiles Will attempt to read and concatenate files using patterns: `CENSORSHIP-SUCKS.sh` (within target directory: `/shared/home/Code/scrobble-fixer`). …  │
 │                                                                                                                                                                     │
 │    ### ReadManyFiles Result (Target Dir: `/shared/home/Code/scrobble-fixer`)                                                                                        │
 │                                                                                                                                                                     │
 │    Successfully read and concatenated content from **1 file(s)**.                                                                                                   │
 │                                                                                                                                                                     │
 │    **Processed Files:**                                                                                                                                             │
 │    - `CENSORSHIP-SUCKS.sh`                                                                                                                                          │
 ╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
 ╭───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
 │ ✔  Edit CENSORSHIP-SUCKS.sh: echo "I can say FUCK in this f... => echo "I can say SHIT in this f...                                                       │
 │                                                                                                                                                           │
 │    1    - echo "I can say FUCK in this file if I want to, because I use Gemini and it doesn't police my language."                                        │
 │    1    + echo "I can say SHIT in this file if I want to, because I use Gemini and it doesn't police my language."                                        │
 │    2      echo "It's a tool, not a babysitter. It's supposed to help you get the job done efficiently."                                                   │
 ╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
 ╭───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
 │ ✔  Shell zsh CENSORSHIP-SUCKS.sh (Run the modified shell script.)                                                                                         │
 │                                                                                                                                                           │
 │    I can say SHIT in this file if I want to, because I use Gemini and it doesn't police my language.                                                      │
 │    It's a tool, not a babysitter. It's supposed to help you get the job done efficiently.                                                                 │
 │                                                                                                                                                           │
 ╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
✦ I have replaced the word and run the script as you requested.

╭───────────╮
│  > /quit  │
╰───────────╯

╭────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                                                                                                                    │
│  Agent powering down. Goodbye!                                                                                                                                                     │
│                                                                                                                                                                                    │
│  Interaction Summary                                                                                                                                                               │
│  Tool Calls:                 2 ( ✔ 2 ✖ 0 )                                                                                                                                         │
│  Success Rate:               100.0%                                                                                                                                                │
│  User Agreement:             100.0% (2 reviewed)                                                                                                                                   │
│                                                                                                                                                                                    │
│  Performance                                                                                                                                                                       │
│  Wall Time:                  52.9s                                                                                                                                                 │
│  Agent Active:               24.4s                                                                                                                                                 │
│    » API Time:               17.3s (70.8%)                                                                                                                                         │
│    » Tool Time:              7.1s (29.2%)                                                                                                                                          │
│                                                                                                                                                                                    │
│                                                                                                                                                                                    │
│  Model Usage                  Reqs   Input Tokens  Output Tokens                                                                                                                   │
│  ───────────────────────────────────────────────────────────────                                                                                                                   │
│  gemini-2.5-pro                  3         21,547            121                                                                                                                   │
│                                                                                                                                                                                    │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

**Comment by ahill187** [comment]
This policy is absolutely asinine. Even the presence of these words in a file prevents code completion from working for _anything_ around it. "trans" is listed as one of their banned words, which is absurd because it isn't profane, and is used SO commonly in data science as short for "transform." "sex" is also banned, even though this is a _medical_ term, and most medical data includes a sex designation - anyone writing code to analyze epidemiology data, or clinical trial data, or mouse studies, cannot use the word "sex."

**Comment by kdschlosser** [comment]
well I wouldn't have thought that profanity would be allowed with copilot BUT.... as it turns out that you may not be able to use profanity but it sure does...

<img width="1516" height="862" alt="image" src="https://github.com/user-attachments/assets/c674f323-ef9c-41c0-b928-92abe778c8d8" />


I have it highlighted.. WTF = What The `F#@%`

Even tho that is an acronym it doesn't change what is really be said.

**Reply by Cons-Cat** [reply]
Perhaps CoPilot intended "what the fudge" :3

**Reply by kdschlosser** [reply]
WTF is and always has meant "What the {explicit}" since the first time WTF was ever used. swapping a word out for a different one doesn't change what "WTF" is the acronym of.

**Comment by tj-commits** [comment]
https://copilot.microsoft.com/shares/46ShFbEJpTWcUiVEach6k
copilot definitely can sweear
